corehq/apps/geospatial/static/geospatial/js/case_grouping_map.js

```diff
@@ -17,11 +17,65 @@ hqDefine("geospatial/js/case_grouping_map",[
         self.coordinates = coordiantes;
         self.caseLink = caseLink;
 
+        // TODO: Group ID needs to be set
+        self.groupId = null;
+
         return self;
     }
 
+    function exportModel() {
+        var self = {};
+
+        self.casesToExport = ko.observableArray([]);
+
+        self.handleExportCSV = function () {
+            const casesToExport = _.map(self.casesToExport(), function (caseItem) {
+                const coordinates = (caseItem.coordinates) ? `${caseItem.coordinates.lng} ${caseItem.coordinates.lat}` : "";
+                return {
+                    'groupId': caseItem.groupId,
+                    'caseId': caseItem.caseId,
+                    'coordinates': coordinates,
+                };
+            });
+            exportToCsv(casesToExport);
+        };
+
+        function exportToCsv(itemsArr, includeHeaders = true) {
+            if (!itemsArr.length) {
+                return;
+            }
+
+            let csvStr = "";
+            if (includeHeaders) {
+                csvStr = Object.keys(itemsArr[0]).join(",");
+                csvStr += "\n";
+            }
+
+            _.forEach(itemsArr, function (itemRow) {
+                csvStr += Object.keys(itemRow).map(key => itemRow[key]).join(",");
+                csvStr += "\n";
+            });
+
+            // Download CSV file
+            const hiddenElement = document.createElement('a');
+            hiddenElement.href = 'data:text/csv;charset=utf-8,' + encodeURI(csvStr);
+            hiddenElement.target = '_blank';
+            hiddenElement.download = `cases_by_group (created on ${getTodayDate()}).csv`;
```

Comment:
nit: is that the filename? I believe we could humanize it a bit. `Grouped cases (${getTodayDate()).csv`

Reply:
It is, I'll update it to make it easier to read. :+1:

Reply:
[3be16e6](https://github.com/dimagi/commcare-hq/pull/33610/commits/3be16e6ffabb113309d086d51bc8763802118060)